project-koku/masu
# Add bucket name directory to contain manifest and CUR files in temporary local storage. Closes #127

Closes #127 
**Overview**
When S3 buckets contain reports of the same name we were overwriting the manifest file in our local temporary storage which would result in the incorrect csv being downloaded which would result in download failures.

The change is to now store the manifest files as well as the .csv files themselves in subdirectories with a name matching the bucket they are coming from.  Because AWS S3 buckets are globally unique we will no longer have any collision issues when saving manifest files.

**Testing**
1. Create 3 local S3 buckets and populate with data (note same report name)
2.  Add the 3 local providers as well as an AWS provider into koku
3. Hit the /api/v1/download endpoint in masu and verify that all reports are downloaded and processed.
4. Verify local directory structure is accurate:

Test Results:
[masu_issue_127.txt](https://github.com/project-koku/masu/files/2267001/masu_issue_127.txt)
